src/en/disasterscans/src/eu/kanade/tachiyomi/extension/en/disasterscans/DisasterScans.kt

```diff
@@ -1,209 +1,133 @@
-    private val preferences: SharedPreferences by lazy {
-        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
-    }
-
-    override fun popularMangaRequest(page: Int): Request {
-        return GET("$apiUrl/comics/search/comics", headers)
-    }
+    private val dateFormat: SimpleDateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.US)
+    private val json by injectLazy<Json>()
 
-    override fun popularMangaParse(response: Response): MangasPage {
-        val comics = response.parseAs<List<ApiSearchComic>>()
+    override fun popularMangaRequest(page: Int): Request =
+        GET("$baseUrl/home", headers)
 
-        val cdnUrl = preferences.getCdnUrl()
+    override fun latestUpdatesRequest(page: Int) =
+        popularMangaRequest(page)
 
-        return MangasPage(comics.map { it.toSManga(cdnUrl) }, false)
-    }
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request =
+        GET("$baseUrl/comics", headers)
 
-    override fun fetchSearchManga(
-        page: Int,
-        query: String,
-        filters: FilterList,
-    ): Observable<MangasPage> {
-        return if (query.startsWith(PREFIX_SLUG)) {
-            val url = "/comics/${query.substringAfter(PREFIX_SLUG)}"
-            val manga = SManga.create().apply { this.url = url }
-            client.newCall(mangaDetailsRequest(manga))
-                .asObservableSuccess()
-                .map { mangaDetailsParse(it).apply { this.url = url } }
-                .map { MangasPage(listOf(it), false) }
-        } else {
-            client.newCall(searchMangaRequest(page, query, filters))
-                .asObservableSuccess()
-                .map { searchMangaParse(it, query) }
-        }
-    }
-
-    override fun searchMangaRequest(page: Int, query: String, filters: FilterList) = popularMangaRequest(page)
+    override fun popularMangaSelector(): String = "div:has(span:contains(POPULAR)) + section a:has(img)"
 
-    private fun searchMangaParse(response: Response, query: String): MangasPage {
-        val comics = response.parseAs<List<ApiSearchComic>>()
+    override fun latestUpdatesSelector(): String = "div:has(span:contains(LATEST)) + section a:has(img)"
 
-        val cdnUrl = preferences.getCdnUrl()
+    override fun searchMangaSelector(): String = ".grid a"
 
-        return comics
-            .filter { it.ComicTitle.contains(query, true) }
-            .map { it.toSManga(cdnUrl) }
-            .let { MangasPage(it, false) }
+    private fun mangaFromElement(element: Element): SManga = SManga.create().apply {
+        setUrlWithoutDomain(element.absUrl("href"))
+        thumbnail_url = element.selectFirst("img")?.absUrl("src")
     }
 
-    override fun mangaDetailsRequest(manga: SManga): Request {
-        return GET("$apiUrl${manga.url}", headers)
+    override fun popularMangaFromElement(element: Element): SManga = mangaFromElement(element).apply {
+        title = element.selectFirst("h5")?.text().toString()
```

Comment:
`.text()` returns String already. Do not use `.toString()` to mask null values. Instead, sinde the title field is critical you can do:
```suggestion
        title = element.selectFirst("h5")!!.text()
```
(not tested)

I see the same pattern being used elsewhere. Search for `.text().toString()` and handle those as well.